pkg/scaler/prober.go

```diff
@@ -459,6 +461,26 @@ func (p *prober) scaleTo(parentContext context.Context, msg string, replicas int
 				klog.V(4).Infof("%s: skipped because desired=%d and current=%d", prefix, replicas, s.Spec.Replicas)
 				continue
 			}
+			/*
+				Check if the scaled objects has defined any delays for the operation.
+				scaleUpDelay is the delay in seconds to wait before initiating scaleUp to ensures that the resource is scaled up after allowing sufficient time for system to recover.
+				scaleDownDelay is the dealy in seconds to wait before initiaing scaleDown to ensure that the resource is scaled down after allowing its dependents room to react.
+			*/
+			// Check for scaleUp delays
+			if replicas > 0 {
+				if dsd.ScaleUpDelay != nil {
+					klog.V(4).Infof("Delaying scale up of %s by %d seconds \n", dsd.ScaleRef.Name, *dsd.ScaleUpDelay)
+					time.Sleep(toDuration(dsd.ScaleUpDelay, 0))
+				}
+			}
+			// Check for scaleDown delays
+			if replicas == 0 {
+				if dsd.ScaleDownDelay != nil {
+					klog.V(4).Infof("Delaying scale down of %s by %d seconds \n", dsd.ScaleRef.Name, *dsd.ScaleUpDelay)
+					time.Sleep(toDuration(dsd.ScaleDownDelay, 0))
+				}
+
+			}
```

Comment:
This could go in an `else` block

Reply:
Done with the[ commit](https://github.com/gardener/dependency-watchdog/pull/39/commits/3277e2204e957df0ede247ce029a40ca148e1749)